bundle/QueryType/QueryCollectionMapper.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Netgen\Bundle\EzPlatformSiteApiBundle\QueryType;
+
+use Netgen\Bundle\EzPlatformSiteApiBundle\View\ContentView;
+use Netgen\Bundle\EzPlatformSiteApiBundle\Search\SortClauseParser;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * @internal
+ *
+ * QueryCollectionMapper maps query configuration from content view to a QueryCollection instance.
+ *
+ * @see \Netgen\Bundle\EzPlatformSiteApiBundle\QueryType\QueryCollection
+ */
+final class QueryCollectionMapper
+{
+    /**
+     * @var \Netgen\Bundle\EzPlatformSiteApiBundle\Search\SortClauseParser
+     */
+    private $sortClauseParser;
+
+    /**
+     * @var \Symfony\Component\HttpFoundation\RequestStack
+     */
+    private $requestStack;
+
+    /**
+     *
+     * @param \Netgen\Bundle\EzPlatformSiteApiBundle\Search\SortClauseParser $sortClauseParser
+     * @param \Symfony\Component\HttpFoundation\RequestStack $requestStack
+     */
+    public function __construct(
+        SortClauseParser $sortClauseParser,
+        RequestStack $requestStack
+    ) {
+        $this->sortClauseParser = $sortClauseParser;
+        $this->requestStack = $requestStack;
+    }
+
+    /**
+     * Map given query $configuration to a QueryCollection instance.
+     *
+     * @param array $configuration
+     * @param \Netgen\Bundle\EzPlatformSiteApiBundle\View\ContentView $view Needed to resolve configured language expressions.
+     *
+     * @return \Netgen\Bundle\EzPlatformSiteApiBundle\QueryType\QueryCollection
+     */
+    public function map(array $configuration, ContentView $view)
+    {
+        $queryCollection = new QueryCollection();
+
+        foreach ($configuration as $name => $queryConfiguration) {
+            $queryDefinition = new QueryDefinition([
+                'name' => $queryConfiguration['query_type'],
+                'parameters' => $this->resolveParameters($queryConfiguration['parameters'], $view),
+                'maxPerPage' => $this->resolveParameter($queryConfiguration['max_per_page'], $view),
+                'page' => $this->resolveParameter($queryConfiguration['page'], $view),
```

Comment:
I hope this geta overriden with `page` query parameter feom the request, so we can actually page :)

Reply:
Depends on the configuration, it does if it was configured like that :)

For example `page: '@=request.query.get("page")'`. This is just resolving the configured value, which means executing language expression if used.

Reply:
Ah, nice. Then it needs to be documented well :)